File: FalconMindViewer/frontend/src/types/mission.ts
export type MissionStatus = 
  | 'draft' 
  | 'scheduled' 
  | 'active' 
  | 'completed' 
  | 'failed' 
  | 'cancelled';

export interface Mission {
  id: string;
  name: string;
  description?: string;
  status: MissionStatus;
  created_by: string;
  assigned_uav_id?: string;
  scheduled_time?: string;
  started_at?: string;
  completed_at?: string;
  parameters: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface MissionCreate {
  name: string;
  description?: string;
  assigned_uav_id?: string;
  scheduled_time?: string;
  parameters?: Record<string, unknown>;
}

File: FalconMindViewer/backend/app/schemas/mission.py
"""
Mission schemas
"""
from typing import Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel


class MissionCreate(BaseModel):
    name: str
    description: Optional[str] = None
    assigned_uav_id: Optional[str] = None
    scheduled_time: Optional[datetime] = None
    parameters: Dict[str, Any] = {}


class MissionUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    assigned_uav_id: Optional[str] = None
    scheduled_time: Optional[datetime] = None
    parameters: Optional[Dict[str, Any]] = None


class MissionStatusUpdate(BaseModel):
    status: str


class MissionResponse(BaseModel):
    id: str
    name: str
    description: Optional[str]
    status: str
    created_by: str
    assigned_uav_id: Optional[str]
    scheduled_time: Optional[str]
    started_at: Optional[str]
    completed_at: Optional[str]
    parameters: Dict[str, Any]
    created_at: str
    updated_at: str

    class Config:
        orm_mode = True
